P1_BMI_Stats.R

```diff
@@ -256,65 +252,399 @@ bmiData <- bmiData %>%
+# 
+# simd_2004 <- read_spss(paste0(lookupFolder,
+#                               "/Unicode/Deprivation/",
+#                               "postcode_2006_2_simd2004.sav")) %>%
+#   select(PC7, simd2004_sc_quintile) %>%
+#   rename(postcode = PC7,
+#          simd = simd2004_sc_quintile) %>%
+#   mutate(year = "simd_2004")
+
+# Assign the appropriate SIMD value to a patient depending on the year they
+# were admitted
+# data <- data %>%
+#   mutate(simd = case_when(
+#     year >= 2014 ~ simd2016_sc_quintile,
+#     year >= 2010 & year <= 2013 ~ simd2012_sc_quintile,
+#     year >= 2007 & year <= 2009 ~ simd2009v2_sc_quintile,
+#     year >= 2004 & year <= 2006 ~ simd2006_sc_quintile,
+#     year <= 2003 ~ simd2004_sc_quintile
+#   )) 
+
+simdMerge <- function(x, y){
+  df <- merge(x, y, by= "pc7", all.x= TRUE, all.y= TRUE)
+  return(df)
+}
+simd <- Reduce(simdMerge, list(simd2016, simd2012, simd2009, simd2006, simd2004))
 
+rescale <- function(x_i){
+  5-x_i
+}
+bmi_data <- bmi_data %>%
+  merge(bmi_data, simd, by = pc7)
+#Recode simd so all go from 1=most to 5=least - this was required for creating allsimdyear/allsimdfinyear----------------*.
+#Still to change labels
+mutate(simd2004_sc_quintile <- sapply(simd2004_sc_quintile,rescale),
+       simd2006_sc_quintile <- sapply(simd2006_sc_quintile,rescale),
+       allsimdfinyear=0)
+
+
+#if(schlyr_exam ==a allsimdfinyear=b)
+bmi_data < bmi_data %>%
+  for (schlyr_exam in c('0102', '0203', '0304', '0405', '0506', '0607','0708', '0809', '0910','1011',
+                        '1112', '1213', '1314', '1415', '1516', '1617', '1718')) {
+    allsimdfinyear <- c('simd2004_sc_quintile', 'simd2004_sc_quintile', 'simd2004_sc_quintile',
+                        'simd2006_sc_quintile', 'simd2006_sc_quintile', 'simd2006_sc_quintile',
+                        'simd2009v2_sc_quintile', 'simd2009v2_sc_quintile','simd2009v2_sc_quintile',
+                        'simd2012_sc_quintile', 'simd2012_sc_quintile', 'simd2012_sc_quintile',
+                        'simd2012_sc_quintile', 'simd2016_sc_quintile', 'simd2016_sc_quintile',
+                        'simd2016_sc_quintile',
+                        'simd2016_sc_quintile')}
+# a= c('0102', '0203', '0304', '0405', '0506', '0607','0708', '0809', '0910','1011', '1112', '1213', '1314',
+#      '1415', '1516', '1617', '1718')
+# b= c('simd2004_sc_quintile', 'simd2004_sc_quintile', 'simd2004_sc_quintile', 'simd2006_sc_quintile',
+#      'simd2006_sc_quintile', 'simd2006_sc_quintile', 'simd2009v2_sc_quintile', 'simd2009v2_sc_quintile',
+#      'simd2009v2_sc_quintile', 'simd2012_sc_quintile', 'simd2012_sc_quintile', 'simd2012_sc_quintile',
+#      'simd2012_sc_quintile', 'simd2016_sc_quintile', 'simd2016_sc_quintile', 'simd2016_sc_quintile',
+#      'simd2016_sc_quintile')
+# bmi_data < bmi_data %>%
+#   for (schlyr_exam in a) {
+#     allsimdfinyear <- b}
+mutate(simd = allsimdfinyear),
+if is.na(allsimdfinyear) simd='U', #x of n records don't have a SIMD quintile.
```

Comment:
do we still need this part? it looks like it won't run so will need to be corrected.